source/data-formats/modeling-bson-data.txt

```diff
@@ -11,140 +11,198 @@ Modeling BSON Data
-classes implementing the :php:`Persistable <mongodb-bson-persistable>` interface
-without requiring the use of the ``typeMap`` option. This is done by encoding
-the name of the PHP class in a special property within the BSON document.
+.. literalinclude:: /includes/bson.php
+    :language: php
+    :dedent:
+    :start-after: start-create-doc
+    :end-before: end-create-doc
 
-.. note::
+Change a BSON Document
+----------------------
 
-   When deserializing a PHP variable from BSON, the encoded class name of a
-   :php:`Persistable <mongodb-bson-persistable>` object will override any class
-   specified in the type map, but it will not override ``"array"`` and
-   ``"stdClass"`` or ``"object"``. This is discussed in the
-   :php:`persistence specification <mongodb.persistence>` but it bears
-   repeating.
+You can modify the contents of a BSON document by using the same notation you use to
+modify an associative array in {+language+}. This example makes the following changes
+to the :ref:`sample BSON document <php-bson-sample>`:
 
-Consider the following class definition:
+- Adds a new ``restaurant_id`` field that has a value of ``12345``
+- Sets the value of the ``name`` field to ``"Mongo's Pizza Place"``
 
-.. code-block:: php
+.. literalinclude:: /includes/bson.php
+    :language: php
+    :dedent:
+    :start-after: start-modify-doc
+    :end-before: end-modify-doc
 
-   <?php
+Customize BSON Serialization
+----------------------------
 
-   class Person implements MongoDB\BSON\Persistable
-   {
-       private MongoDB\BSON\ObjectId $id;
-       private string $name;
-       private MongoDB\BSON\UTCDateTime $createdAt;
-
-       public function __construct(string $name)
-       {
-           $this->id = new MongoDB\BSON\ObjectId;
-           $this->name = $name;
-           $this->createdAt = new MongoDB\BSON\UTCDateTime;
-       }
-
-       function bsonSerialize()
-       {
-           return [
-               '_id' => $this->id,
-               'name' => $this->name,
-               'createdAt' => $this->createdAt,
-           ];
-       }
-
-       function bsonUnserialize(array $data)
-       {
-           $this->id = $data['_id'];
-           $this->name = $data['name'];
-           $this->createdAt = $data['createdAt'];
-       }
-   }
+The following sections describe how to configure your application's 
+serialization of BSON data:
 
-The following example constructs a ``Person`` object, inserts it into the
-database, and reads it back as an object of the same type:
+- :ref:`php-type-map`: Use the ``typeMap`` option to specify the default conversion
+  between {+language+} types and BSON types
+
+- :ref:`php-persistable-classes`: Use the ``MongoDB\BSON\Persistable`` interface to
+  enable serialization
+
+- :ref:`php-enums`: Use the ``bsonSerialize()`` and ``bsonUnserialize()`` methods to
+  specify serialization between backed enums and BSON values
+
+.. _php-type-map:
```

Comment:
S: consider adding bson to this anchor unless its preserved from a prev version of this doc
```suggestion
.. _php-bson-type-map:
```